DashAI/back/dataloaders/classes/csv_dataloader.py

```diff
@@ -0,0 +1,35 @@
+import io
+import os
+import zipfile
+
+from datasets import load_dataset
+
+from DashAI.back.dataloaders.classes.dataloader import DataLoader
+
+
+class CSVDataLoader(DataLoader):
+    """
+    Data loader for tabular data in CSV files
+    """
+
+    def load_data(self, dataset_path, params, file=None, url=None):
+        separator = params["separator"]
```

Comment:
This dictionary of parameters is defined by the pydantic model 'DatasetParams', where you have the specific parameters of the dataloaders, among which is the separator for the case of the csv dataloader.
```
class DataLoaderParams(BaseModel):
    separator: Optional[str] = ","
    data_key: Optional[str] = "data"
    # add other params of dataloaders
...
class DatasetParams(BaseModel):
    task_name: str
    data_loader: str
    dataset_name: str
    class_column: Union[int, str] = -1
    splits_in_folders: bool = False
    splits: SplitParams
    dataloader_params: DataLoaderParams
```
So in the endpoint to upload datasets the key 'dataloader_params' that contain these parameters is called to pass the parameters to the dataloader in 'params' argument.
```
dataset = dataloader.load_data(
            dataset_path=folder_path,
            params=params.dataloader_params.dict(),
            file=file,
            url=url,
        )
```
When the input string with the parameters is parsed to the pydantic model, automatically the parameters that defines the model that do not exist in the parameters input are added, so the diccionary of parameter always will have all the keys with default values, even when one is not present at the user's input.
For that it is assumed that the key exists, but I will add the check anyways because is necessary that exist.

Reply:
Thanks.